nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5398,6 +6103,43 @@ async fn vpc_router_update(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// List routes
+///
+/// List the routes associated with a router in a particular VPC.
+#[endpoint {
+    method = GET,
+    path = "/v1/vpc-router-routes",
+    tags = ["vpcs"],
+}]
+async fn vpc_router_route_list_v1(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    query_params: Query<PaginatedByNameOrId<params::RouterSelector>>,
+) -> Result<HttpResponseOk<ResultsPage<RouterRoute>>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let nexus = &apictx.nexus;
+        let query = query_params.into_inner();
+        let pag_params = data_page_params_for(&rqctx, &query)?;
+        let scan_params = ScanByNameOrId::from_query(&query)?;
+        let paginated_by = name_or_id_pagination(&pag_params, scan_params)?;
+        let router_lookup =
+            nexus.vpc_router_lookup(&opctx, &scan_params.selector)?;
+        let routes = nexus
+            .vpc_router_route_list(&opctx, &router_lookup, &paginated_by)
+            .await?
+            .into_iter()
+            .map(|route| route.into())
+            .collect();
+        Ok(HttpResponseOk(ScanByNameOrId::results_page(
+            &query,
+            routes,
+            &marker_for_name_or_id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
 // Vpc Router Routes
 
 /// List routes
```

Comment:
I think this method needs `deprecated = true` and a reference to the new API to use.